logging/logging.go

```diff
@@ -0,0 +1,108 @@
+package logging
+
+import (
+	"log/slog"
+	"net/http"
+	"strings"
+	"time"
+
+	azcorePolicy "github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+)
+
+var resourceTypes = map[string]bool{
+	"resourcegroups":   true,
+	"storageaccounts":  true,
+	"operationresults": true,
+	"asyncoperations":  true,
+}
+
+type LogRequestParams struct {
+	Logger    *slog.Logger
+	StartTime time.Time
+	Request   interface{}
+	Response  *http.Response
+	Error     error
+}
+
+// Shared logging function for REST API interactions
```

Comment:
Use AI to add document for public functions

Reply:
Used copilot to modify readme to add info about LogRequest function.